src/index.js

```diff
@@ -0,0 +1,134 @@
+'use babel';
+
+// eslint-disable-next-line import/extensions, import/no-extraneous-dependencies
+import { CompositeDisposable } from 'atom';
+import path from 'path';
+import escapeHtml from 'escape-html';
+import pluralize from 'pluralize';
+import * as helpers from 'atom-linter';
+
+const DEFAULT_ARGS = [
+  '--cache', 'false',
+  '--force-exclusion',
+  '--format', 'json',
+  '--display-style-guide',
+];
+
+const formatMessage = ({ message, cop_name: copName, url }) => {
+  const formattedMessage = escapeHtml(message || 'Unknown Error');
+  let formattedCopName = '';
+  if (copName && url) {
+    formattedCopName = ` (<a href="${escapeHtml(url)}">${escapeHtml(copName)}</a>)`;
+  } else if (copName) {
+    formattedCopName = ` (${escapeHtml(copName)})`;
+  }
+  return formattedMessage + formattedCopName;
+};
+
+const parseFromStd = (stdout, stderr) => {
+  let parsed;
+  try {
+    parsed = JSON.parse(stdout);
+  } catch (error) {
+    // continue regardless of error
+  }
+  if (typeof parsed !== 'object') { throw new Error(stderr || stdout); }
+  return parsed;
+};
+
+const forwardRubocopToLinter =
+  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
+    const [message, url] = rawMessage.split(/ \((.*)\)/, 2);
+    const { line, column, length } = location;
+    const outputToLinter = {
+      type: ['refactor', 'convention', 'warning'].includes(severity) ? 'Warning' : 'Error',
+      html: formatMessage({ cop_name, message, url }),
+      filePath,
+    };
+    return Object.assign(outputToLinter,
+      location && { range: [[line - 1, column - 1], [line - 1, (column + length) - 1]] },
```

Comment:
Although this is nice and compact, wouldn't this throw a `TypeError` if location was undefined (since it would then be `false` and not an `Object`)?

I was thinking something simpler, along the lines of:

```js
if (location) {
  outputToLinter.range = [[line - 1, column - 1], [line - 1, (column + length) - 1]];
}
return outputToLinter;
```